R/method-correlation.R

```diff
@@ -262,34 +242,154 @@ buildCorrelationComputeResult <- function(corrResult, data1, data2 = NULL, metho
+
+  veupathUtils::logWithTime(paste("Received df table with", nrow(values), "samples and", (ncol(values)-1), "features with values."), verbose)
+
+  if (format == 'data.table') {
+    return(corrResult)
+  } else {
+    result <- buildCorrelationComputeResult(corrResult, data, NULL, method, verbose)
+    result@computationDetails <- 'selfCorrelation'
+    return(result)
+  }  
+})
+
+#' @rdname selfCorrelation
+#' @aliases selfCorrelation,SampleMetadata-method
+setMethod("selfCorrelation", signature("SampleMetadata"), 
+function(data, method = c('spearman','pearson'), format = c('ComputeResult', 'data.table'), verbose = c(TRUE, FALSE)) {
+
+  format <- veupathUtils::matchArg(format)
+  method <- veupathUtils::matchArg(method)
+  verbose <- veupathUtils::matchArg(verbose)
+  
+  corrResult <- correlation(getSampleMetadata(data, TRUE, FALSE), NULL, method = method, format = 'data.table', verbose = verbose)
+
+  veupathUtils::logWithTime(paste("Received df table with", nrow(data), "samples and", (ncol(data)-1), "variables."), verbose)
+
+  if (format == 'data.table') {
+    return(corrResult)
+  } else {
+    result <- buildCorrelationComputeResult(corrResult, data, NULL, method, verbose)
+    result@computationDetails <- 'selfCorrelation'
+    return(result)
+  }
+})
+
+#' @rdname correlation
+#' @aliases correlation,Collection,Collection-method
+setMethod("correlation", signature("Collection", "Collection"), 
+function(data1, data2, method = c('spearman','pearson'), format = c('ComputeResult', 'data.table'), verbose = c(TRUE, FALSE), proportionNonZeroThreshold = 0.5, varianceThreshold = 0, stdDevThreshold = 0) {
+  
+  format <- veupathUtils::matchArg(format)
+  method <- veupathUtils::matchArg(method)
+  verbose <- veupathUtils::matchArg(verbose)
+  
+  #prefilters applied
+  data1 <- pruneFeatures(data1, predicateFactory('proportionNonZero', proportionNonZeroThreshold), verbose)
+  data1 <- pruneFeatures(data1, predicateFactory('variance', varianceThreshold), verbose)
+  data1 <- pruneFeatures(data1, predicateFactory('sd', stdDevThreshold), verbose)
+  data2 <- pruneFeatures(data2, predicateFactory('proportionNonZero', proportionNonZeroThreshold), verbose)
+  data2 <- pruneFeatures(data2, predicateFactory('variance', varianceThreshold), verbose)
+  data2 <- pruneFeatures(data2, predicateFactory('sd', stdDevThreshold), verbose)
+  
+  values1 <- getCollectionData(data1, FALSE, TRUE, verbose)
+  values2 <- getCollectionData(data2, FALSE, TRUE, verbose)
+
+  # empty samples removed from data by getCollectionData, means we need to keep samples common to both datasets and remove ids
+  # get id col names
+  recordIdColumn <- data1@recordIdColumn
+  allIdColumns <- c(recordIdColumn, data1@ancestorIdColumns)
+  # should we verify that ids are the same in both datasets?
+
+  # remove samples that are not common
+  commonSamples <- intersect(values1[[recordIdColumn]], values2[[recordIdColumn]])
```

Comment:
we should also log that we do this. A "hey btw we're moving forward with X number of samples because that's what we found overlapping in the two datasets"